src/plugins/api/index_pattern_endpoints.js

```diff
@@ -0,0 +1,221 @@
+const esErrors = require('elasticsearch').errors;
+const Boom = require('Boom');
+const _ = require('lodash');
+const Promise = require('bluebird');
+const getMappings = require('./lib/get_mappings');
+const stitchPatternAndMappings = require('./lib/stitch_pattern_and_mappings');
+const {templateToPattern, patternToTemplate} = require('./lib/convert_pattern_and_template_name');
+const removeDeprecatedFieldProps = require('./lib/remove_deprecated_field_props');
+
+export default function (server) {
+
+  let handleESError = function (error) {
+    if (error instanceof esErrors.ConnectionFault ||
+      error instanceof esErrors.ServiceUnavailable ||
+      error instanceof esErrors.NoConnections ||
+      error instanceof esErrors.RequestTimeout) {
+      return Boom.serverTimeout();
+    } else if (error instanceof esErrors.Conflict) {
+      return Boom.conflict();
+    } else if (error instanceof esErrors[403]) {
+      return Boom.forbidden();
+    } else if (error instanceof esErrors.NotFound) {
+      return Boom.notFound();
+    } else if (error instanceof esErrors.BadRequest || error instanceof TypeError) {
+      return Boom.badRequest(error);
+    } else {
+      return error;
+    }
+  };
+
+  server.route({
+    path: '/api/index-patterns',
+    method: 'GET',
+    handler: function (req, reply) {
+      let client = server.plugins.elasticsearch.client;
```

Comment:
You probably want to use `callWithRequest` here, so 401s are properly forwarded to the user. Right @simianhacker?